homeassistant/components/hue/sensor_base.py

```diff
@@ -140,13 +139,14 @@ def async_update_items(self):
             primary_sensor = primary_sensor_devices.get(_device_id(api[item_id]))
             if primary_sensor is not None:
                 base_name = primary_sensor.name
-            name = sensor_config["name_format"].format(base_name)
 
-            current[uniqueid] = sensor_config["class"](
-                api[item_id], name, self.bridge, primary_sensor=primary_sensor
-            )
+            for sensor in sensor_config:
+                name = sensor["name_format"].format(base_name)
+                current[uniqueid] = sensor["class"](
```

Comment:
Don't we override the key here since `uniqueid` remains the same?

Reply:
We can keep the unique identifier because the HueBattery class adds the suffix "-battery"

    @property
    def unique_id(self):
        """Return a unique identifier for this device."""
        return f"{self.sensor.uniqueid}-battery"

The register of entities thus stores two separate identifiers

                "supported_features": 0,
                "unique_id": "00:17:88:01:03:XX:XX:XX-02-0406-battery",
                "unit_of_measurement": "%"
            },

Reply:
That suffix is added inside the entity in the `unique_id` property. It doesn't affect `uniqueid` variable here. So we are overwriting the same key.

We need to either extend the key or store the entities in a list under the key.

Side note: Storing the entities is a bad pattern and not really needed. We should use the dispatch helper to signal the entities instead and just use a set to store the ids for the entities.

Reply:
On Hue Bridge, the motion sensor is actually three separate sensors and the battery level is used up on each.
motion: 00: 17: 88: 01: 04: XX: XX: XX-02-0406
lighting level: 00: 17: 88: 01: 04: XX: XX: XX-02-0400
temperature: 00: 17: 88: 01: 04: XX: XX: XX-02-0402
For the battery sensor, I suggest you make a regular expression and only take the start 00: 17: 88: 01: 04: XX: XX: XX as the unique identifier if class is Battery
What do you think?
pattern : ([0-9a-fA-F]:?){16}

This would also apply to dimmers with a HueBattery class

Reply:
We can do it without a regex and just follow unique ID? The only thing is that we cannot change the unique ID of existing entities that users have already used, or else there will be ghost entities.

Reply:
> We can do it without a regex and just follow unique ID? The only thing is that we cannot change the unique ID of existing entities that users have already used, or else there will be ghost entities.

I would do as @MartinHjelmare suggests and store a list of entities in each `current[uniqueid]`, as it is the quickest fix without moving to another pattern there.

It is no problem to do `current[uniqueid] = [sensor["class"](...) for sensor in sensor_config]` in that discovery method, and it would not change any old uniqueid if just the new batt sensor in ZLLSwitch is made with `uniqueid-battery`